tvl_backends/tvl-backends-fffr/ext/bindings/TvFFFrameReader.cpp

```diff
@@ -175,39 +177,36 @@ uint8_t* TvFFFrameReader::read_frame()
 
 int64_t TvFFFrameReader::read_frames_by_index(int64_t* indices, const int n_frames, uint8_t** frames)
 {
-    int32_t block_size = 8;
-    for (int32_t pos = 0; pos < n_frames;) {
+    int32_t pos = 0;
+    for (; pos < n_frames;) {
         const auto start = &indices[pos];
-        const auto last = pos + block_size;
+        const auto last = pos + static_cast<int32_t>(bufferLength);
         const auto end = &indices[std::min(last, n_frames)];
         const std::vector<int64_t> frame_sequence(start, end);
         const auto frames_vector = _stream->getFramesByIndex(frame_sequence);
-        int32_t n_frames_read = frames_vector.size();
-        // If the actual number of frames decoded is less than the number of frames that we
-        // requested, reduce the number of frames that we request next time and ignore the last
-        // frame returned (it might be wrong).
-        if (n_frames_read < frame_sequence.size()) {
-            block_size = n_frames_read;
-            --n_frames_read;
```

Comment:
Can you now guarantee that all frames returned when `n_frames_read < frame_sequence.size()` will be correct? I don't want to start getting bad frames returned, as that would be extremely difficult to track down.

Reply:
Yep, fffr should never return bad frames.